reports.js

```diff
@@ -38,7 +38,41 @@ async function pluginsReport() {
     report.push(plugin);
   });
   report.sort((a, b) => b.installs - a.installs);
-  return report;
+
+  const reducer = (accumulator, currentValue) => accumulator + currentValue;
+  const status = {
+    todo: [],
+    PR: [],
+    done: [],
+    total:[]
+  }
+  const byStatus = report.forEach(plugin => {
+    switch (plugin.status) {
+      case 'TODO':
+        status.todo.push(1);
+        break;
+      case 'PR':   
+        status.PR.push(1);
+        break;
+      case 'OK':
+        status.done.push(1);
+        break;
+      default:
+        throw new Error('Unknown type');
+    }
+  })
+
+  const todo = status.todo.reduce(reducer);
```

Comment:
Reduce actually will run once for each element of the are array. Passing the second parameter (undefined) as the initial accumulator.

Really status.todo.length is what you want since it's the number elements in the array.

Or in your for each, instead of adding an element to the array, just start at 0 and add 1 each time.

Total can also just be report.lenght, unless your thinking there are statuses we will ignore.